plugin/src/main/groovy/com/novoda/buildproperties/BuildPropertiesPlugin.groovy

```diff
@@ -28,69 +28,84 @@ class BuildPropertiesPlugin implements Plugin<Project> {
-        }
-        target.ext.buildConfigProperty = { String name, Entry entry ->
-            project.afterEvaluate {
-                target.buildConfigField 'String', name, "\"${entry.string}\""
-            }
+        target.ext.buildConfigBoolean = { String name, def value ->
+            buildConfigField('boolean', name, {
+                return Boolean.toString(value instanceof Entry ? value.boolean : value)
+            })
+        }
+        target.ext.buildConfigInt = { String name, def value ->
+            buildConfigField('int', name, {
+                return Integer.toString(value instanceof Entry ? value.int : value)
+            })
+        }
+        target.ext.buildConfigLong = { String name, def value ->
+            buildConfigField('long', name, {
+                return "${Long.toString(value instanceof Entry ? value.long : value)}L"
+            })
+        }
+        target.ext.buildConfigDouble = { String name, def value ->
+            buildConfigField('double', name, {
+                return Double.toString(value instanceof Entry ? value.double : value)
+            })
+        }
+        target.ext.buildConfigString = { String name, def value ->
+            buildConfigField('String', name, {
+                return "\"${value instanceof Entry ? value.string : value}\""
+            })
         }
     }
 
-    private static void addResValueSupportTo(target, Project project) {
+    private static void addResValueSupportTo(target) {
         def resValue = { String type, String name, Closure<String> value ->
             target.resValue type, name, value()
         }
-        target.ext.resValueBoolean = { String name, Boolean value ->
-            resValue('bool', name, { Boolean.toString(value) })
-        }
-        target.ext.resValueInt = { String name, Integer value ->
-            resValue('integer', name, { Integer.toString(value) })
-        }
-        target.ext.resValueString = { String name, String value ->
-            resValue('string', name, { "\"$value\"" })
-        }
-        target.ext.resValueProperty = { String name, Entry entry ->
-            project.afterEvaluate {
-                target.resValue 'string', name, "\"${entry.string}\""
-            }
+        target.ext.resValueBoolean = { String name, def value ->
+            resValue('bool', name, {
+                if (value instanceof Entry) {
+                    return value.boolean
+                }
+                return Boolean.toString(value)
+            })
+        }
+        target.ext.resValueInt = { String name, def value ->
+            resValue('integer', name, {
+                if (value instanceof Entry) {
+                    return value.int
+                }
+                return Integer.toString(value)
+            })
+        }
+        target.ext.resValueString = { String name, def value ->
+            resValue('string', name, {
+                if (value instanceof Entry) {
```

Comment:
I spoke to Toto and I was going to push:

```groovy
    private static void addResValueSupportTo(target) {
        def resValue = { String type, String name, Closure<String> value ->
            target.resValue type, name, value()
        }
        target.ext.resValueBoolean = { String name, def value ->
            resValue('bool', name, {
                return Boolean.toString(value instanceof Entry ? value.boolean : value)
            })
        }
        target.ext.resValueInt = { String name, def value ->
            resValue('integer', name, {
                return Integer.toString(value instanceof Entry ? value.int : value)
            })
        }
        target.ext.resValueString = { String name, def value ->
            resValue('string', name, {
                return "\"${value instanceof Entry ? value.string : value}\""
            })
        }
    }
```

but the tests still passed if I put `return Integer.toString(value instanceof Entry ? value.boolean : value)` for example, so probably missing some tests.

Will let him update :grin: